client/benches/tps/oneshot.rs

```diff
@@ -2,10 +2,41 @@
 
 mod lib;
 
-#[allow(clippy::expect_used)]
+use std::{fs::File, io::BufWriter};
+
+use tracing_flame::{FlameLayer, FlushGuard};
+use tracing_subscriber::prelude::*;
+
+#[allow(clippy::expect_used, clippy::print_stdout, clippy::use_debug)]
 fn main() {
+    let args: Vec<String> = std::env::args().collect();
+    let mut flush_guard: Option<FlushGuard<BufWriter<File>>> = None;
+
+    if args.len() >= 2 {
+        let file = File::create(&args[1]).expect("valid path");
+
+        let flame_layer = FlameLayer::new(BufWriter::new(file))
+            .with_threads_collapsed(true)
+            .with_empty_samples(true);
+        flush_guard = Some(flame_layer.flush_on_drop());
+
+        tracing_subscriber::registry().with(flame_layer).init();
+        iroha_logger::disable_logger();
+    }
+
     let config = lib::Config::from_path("benches/tps/config.json").expect("Failed to configure");
     let tps = config.measure().expect("Failed to measure");
-    iroha_logger::info!(?config);
-    iroha_logger::info!(%tps);
+
+    match flush_guard {
+        Some(guard) => {
+            guard.flush().expect("Flushed data without errors");
```

Comment:
We don't need to manually call `flush`, it will [call it on drop](https://docs.rs/tracing-flame/0.2.0/src/tracing_flame/lib.rs.html#363).

Reply:
I think it's better to leave it explicit. That way when someone reads this code they are aware of what is happening.

Reply:
@SamHSmith, don't we call `flush` 2 times with this implementation?